proxy/src/console/messages.rs

```diff
@@ -0,0 +1,190 @@
+use serde::Deserialize;
+use std::fmt;
+
+/// Generic error response with human-readable description.
+/// Note that we can't always present it to user as is.
+#[derive(Debug, Deserialize)]
+pub struct ConsoleError {
+    pub error: Box<str>,
+}
+
+/// Response which holds client's auth secret, e.g. [`crate::scram::ServerSecret`].
+/// Returned by the `/proxy_get_role_secret` API method.
+#[derive(Deserialize)]
+pub struct GetRoleSecret {
+    pub role_secret: Box<str>,
+}
+
+// Manually implement debug to omit sensitive info.
+impl fmt::Debug for GetRoleSecret {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("GetRoleSecret").finish_non_exhaustive()
+    }
+}
+
+/// Response which holds compute node's `host:port` pair.
+/// Returned by the `/proxy_wake_compute` API method.
+#[derive(Debug, Deserialize)]
+pub struct WakeCompute {
+    pub address: Box<str>,
+    pub aux: MetricsAuxInfo,
+}
+
+/// Async response which concludes the link auth flow.
+/// Also known as `kickResponse` in the console.
+#[derive(Debug, Deserialize)]
+pub struct KickSession<'a> {
+    /// Session ID is assigned by the proxy.
+    pub session_id: &'a str,
+
+    /// Compute node connection params.
+    #[serde(deserialize_with = "KickSession::parse_db_info")]
+    pub result: DatabaseInfo,
+}
+
+impl KickSession<'_> {
+    fn parse_db_info<'de, D>(des: D) -> Result<DatabaseInfo, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        enum Wrapper {
+            // Currently, console only reports `Success`.
+            // `Failure(String)` used to be here... RIP.
+            Success(DatabaseInfo),
```

Comment:
Looks ok to me. FYI @neondatabase/console

Reply:
Not sure what "Currently, console only reports `Success`." means. The console can respond [both with error and with successful responses](https://github.com/neondatabase/cloud/blob/360de29761e0e00976f5bd449409adbd843fd627/openapispec/v2/management-v2.yaml#L263-L271) in the `  /proxy_wake_compute` endpoint. So what "Failure" means here?

Reply:
Good question! This comment refers to the previous definition [over here](https://github.com/neondatabase/neon/pull/3211/files/b7c08055964c7342b23a378a3f013ae1c97cb57e#diff-bc748f50c64625a1b159dbb51dbfb43eaebba5f929298dff9cc03b3ee697a8a5L53-L92
). This is known as `kickResponse` in the console and is used in the link auth. You may notice that previously we had

```rust
#[derive(Debug, Deserialize)]
enum PsqlSessionResult {
    Success(DatabaseInfo),
    Failure(String),
}
```

However, I haven't been able to find any trace of `Failure` in the console code. It occurs to me that this has never been utilized, which is why I've simplified that part and dropped this variant altogether. Please correct me if I'm wrong.

Reply:
@antonyc sorry, I accidentally deleted your comment.

Reply:
> Please correct me if I'm wrong.

I don't fully understand how the spec would to support both String and DatabaseInfo. So for now only DatabaseInfo can be returned.